sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/com/azure/resourcemanager/eventgrid/EventGridPublishAndConsumeExample.java

```diff
@@ -0,0 +1,234 @@
+import com.azure.resourcemanager.eventgrid.models.EventSubscriptionFilter;
+import com.azure.resourcemanager.eventgrid.models.Topic;
+import com.azure.resourcemanager.eventhubs.EventHubsManager;
+import com.azure.resourcemanager.eventhubs.models.EventHub;
+import com.azure.resourcemanager.eventhubs.models.EventHubNamespace;
+import com.azure.resourcemanager.eventhubs.models.EventHubNamespaceSkuType;
+import com.azure.resourcemanager.resources.models.ResourceGroup;
+import reactor.core.publisher.Flux;
+
+import java.util.Random;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static com.azure.messaging.eventhubs.implementation.ClientConstants.OPERATION_TIMEOUT;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class EventGridPublishAndConsumeExample {
+    private static AzureResourceManager resourceManager;
+    private static EventGridManager eventGridManager;
+    private static EventHubsManager eventHubsManager;
+
+    private static final Random RANDOM = new Random();
+    private static final int NUMBER_OF_EVENTS = 10;
+    private static final Region region = Region.US_WEST2;
+    private static final String resourceGroupName = "rg" + randomPadding();
+    private static final String eventHubName = "eh" + randomPadding();
+    private static final String eventHubNamespace = "eh-namespace" + randomPadding();
+    private static final String topicName = "my-topic-name" + randomPadding();
+    private static final String eventSubscriptionName = "event-subscription" + randomPadding();
+    private static final String eventHubRuleName = "my-management-rule" + randomPadding();
+
+    /**
+     * Main entry point.
+     *
+     * @param args the parameters
+     */
+    public static void main(String[] args) {
+        try {
+            // 1. Authenticate
+            // Authentication environment variables need to be set: AZURE_CLIENT_ID,AZURE_TENANT_ID,AZURE_CLIENT_SECRET,AZURE_SUBSCRIPTION_ID
+            TokenCredential credential = new EnvironmentCredentialBuilder()
+                .authorityHost(AzureAuthorityHosts.AZURE_PUBLIC_CLOUD)
+                .build();
+
+            AzureProfile profile = new AzureProfile(AzureEnvironment.AZURE);
+
+            // 2. Create ResourceManager, EventHubsManager, EventGridManager
+            resourceManager = AzureResourceManager.configure()
+                .withLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS)
+                .authenticate(credential, profile)
+                .withDefaultSubscription();
+
+            eventHubsManager = EventHubsManager.configure()
+                .withLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS)
+                .authenticate(credential, profile);
+
+            eventGridManager = EventGridManager.configure()
+                .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS))
+                .authenticate(credential, profile);
```

Comment:
`EventHubsManager` seems not needed? `AzureResourceManager` should contains it.

Also you can see if they can share a same `HttpClient` instance.

Reply:
updated